opm/grid/cpgrid/processEclipseFormat.cpp

```diff
@@ -668,9 +680,24 @@ namespace cpgrid
 
             // Build face to cell.
             f2c.clear();
-            int nf = output.number_of_faces;
             cpgrid::EntityRep<0> cells[2];
             face_to_output_face.clear();
+            // Add nnc connections first, to preserve the
+            // property that top and bottom faces come last
+            // (per cell) in the cell-face mapping.
+            for (const auto& nncpair : nnc) {
+                const int c1 = global_to_local[nncpair.first];
+                const int c2 = global_to_local[nncpair.second];
+                if (c1 < 0 || c2 < 0) {
+                    OPM_THROW(std::runtime_error, "NNC connection requested between inactive cells.");
+                }
+                cells[0].setValue(c1, true);
```

Comment:
Here you silently assume that an NNC between to cells may only occur once.
This will not work in general, because:

1. It is totally legal to specify an NNC between already connected cells.
2. It is totally legal to specify the same NNC multiple times. In that case transmissibilities will be added.
3. A specified NNC might also occur from PINCH.

Reply:
You are correct, I will amend to make sure it works in those three cases.

Reply:
I tried to understand why the test of the companion PR in ewoms fails.

Could it be that we now end up with two intersections in case an NNC was created via PINCH? From the commented code below it seems that there is an intersection between the old cell pairs (before pinching) that is never removed. Instead we now add a new intersection. Hence we should have one intersection with the old (now inactive cell, that hopefully gets zero transmissibility later) and another one with the correct cell.